libs/apigateway-lambda-proxy/src/lib/metric-alarms.ts

```diff
@@ -0,0 +1,376 @@
+import * as pulumi from '@pulumi/pulumi'
+import * as aws from '@pulumi/aws'
+
+const namespace = 'AWS/ApiGateway'
+
+/**
+ * Dimensions to identify an HTTP API in CloudWatch.
+ *
+ * The main difference is REST APIs are identified by name, while HTTP APIs are identified by id.
+ */
+export type HttpGateway = {
+    /**
+     * The API Gateway ID to monitor. Only valid for HTTP endpoints.
+     */
+    id: pulumi.Input<string>
+
+    /**
+     * The API name to monitor.
+     *
+     * For REST endpoints, this must be the name of the API created in AWS.
+     * For HTTP endpoints, this can be anything (it's only used in alerts).
+     */
+    name: pulumi.Input<string> | undefined
+
+    /**
+     * The stage to monitor
+     */
+    stage?: pulumi.Input<string>
+}
+
+/**
+ * Dimensions to identify an API in CloudWatch.
+ */
+export type ApiGateway = HttpGateway
+
+/**
+ * Provides an opinionated set of recommended alarms for an API Gateway stage.
+ *
+ * Raise alarms for the following:
+ *  - 5xx error rate exceeded (default: 2%)
+ *  - 4xx error rate exceeded (default: 50%)
```

Comment:
This seems pretty high? What are these numbers based on?

Reply:
4xx means client-side error, which means it's usually not actionable.

Reply:
This might not ever trigger as the majority of our traffic is GET. Can we target the metric to POST requests only as that would be more likely to catch real issues.